.github/workflows/nightly-release-test.yml

```
@@ -10,7 +10,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu, macos]
-        python-version: ["3.11", "3.12", "3.13", "3.14"]
+        python-version: ["3.11", "3.12", "3.13", "3.14.0"]
```

I assume the purpose of this PR is to have the tests on open PRs running again and ease reviewing, but I think we should let the nightly fail as they are more release-oriented. Also, keeping the nightly tests active would help us and downstream package maintainers quickly know when the dataclass error is fixed.
```suggestion
        python-version: ["3.11", "3.12", "3.13", "3.14"]
```

from [spec4](https://scientific-python.org/specs/spec-0004/):

> Regularly running your project’s tests while using the nightly version of your dependencies allows you to spot problems caused by upstream changes before a new release is made. This way potential issues can be resolved before they find their way into a release, at which point it becomes much harder to change or revert something.

> Regularly creating nightly wheels for your project allows projects that depend on you to give feedback about upcoming changes. As with testing against nightlies of your dependencies this gives your dependents a chance to report problems before they find their way into a release.